src/test/java/com/suse/salt/netapi/calls/modules/FileTest.java

```diff
@@ -392,4 +462,68 @@ public final void testRmdir() {
         assertEquals(new JsonPrimitive(errorMessage),
                 ((JsonParsingError) response.get("minion1").error().get()).getJson());
     }
+
+    @Test
+    public final void testIsLink() {
+        // true response
+        stubFor(any(urlMatching("/"))
+                .willReturn(aResponse()
+                        .withStatus(HttpURLConnection.HTTP_OK)
+                        .withHeader("Content-Type", "application/json")
+                        .withBody(JSON_TRUE_RESPONSE)));
+
+        LocalCall<Boolean> call = File.isLink("/test/");
+        assertEquals("file.is_link", call.getPayload().get("fun"));
+
+        Map<String, Result<Boolean>> response = call.callSync(client,
+                new MinionList("minion1"), AUTH).toCompletableFuture().join();
+        assertTrue(response.get("minion1").result().get());
+
+        // false response
+        stubFor(any(urlMatching("/"))
+                .willReturn(aResponse()
+                        .withStatus(HttpURLConnection.HTTP_OK)
+                        .withHeader("Content-Type", "application/json")
+                        .withBody(JSON_FALSE_RESPONSE)));
+
+        call = File.isLink("/test/");
+        assertEquals("file.is_link", call.getPayload().get("fun"));
+
+        response = call.callSync(client,
+                new MinionList("minion1"), AUTH).toCompletableFuture().join();
+        assertFalse(response.get("minion1").result().get());
+
+    }
+
+    @Test
+    public final void testSymlink() {
+        // true response
+        stubFor(any(urlMatching("/"))
+                .willReturn(aResponse()
+                        .withStatus(HttpURLConnection.HTTP_OK)
+                        .withHeader("Content-Type", "application/json")
+                        .withBody(JSON_TRUE_RESPONSE)));
+
+        LocalCall<Boolean> call = File.symlink("/test/", "/link");
+        assertEquals("file.symlink", call.getPayload().get("fun"));
+
+        Map<String, Result<Boolean>> response = call.callSync(client,
+                new MinionList("minion1"), AUTH).toCompletableFuture().join();
+        assertTrue(response.get("minion1").result().get());
+
+        // false response
```

Comment:
For all these tests I would suggest to separate the tests for the `true` and `false` responses, and rather create one test method for each use-case. This ensures that all assertions are actually performed. Otherwise if the first assertion is failing, the second one would not even be executed. Also if this test for example would fail, you don't see directly which of the use-cases is broken (the `true` or the `false` response).

Reply:
I agree with you, I think each case should be in a separated method and that was my first approach. 
The reason behind my change of approach was to keep the consistency with the rest of the project test, with one test method per function with several scenarios.
However in this case following the right way instead of consistency seems not to bring any harm to the future of the project, therefore I will  split all the tests into suggested into one per scenario.